app/src/shared/components/PasswordStrength/index.jsx

```diff
@@ -0,0 +1,86 @@
+// @flow
+
+import { PureComponent, type Node } from 'react'
+
+import CancelledPromiseError from '$shared/errors/CancelledPromiseError'
+import makeCancelable, { type Cancelable } from '$utils/makeCancelable'
+import zxcvbn from '$utils/zxcvbn'
+
+type Props = {
+    children?: (number) => Node,
+    enabled?: boolean,
+    value?: string,
+}
+
+type State = {
+    measurer: ?(string) => {
+        score: number,
+    },
+    strength: number,
+}
+
+class PasswordStrength extends PureComponent<Props, State> {
+    state = {
+        measurer: null,
+        strength: -1,
+    }
+
+    componentDidMount() {
+        this.loadMeasurer()
+    }
```

Comment:
Also when when this mounts, it goes on a real journey:

1. `componentDidMount` calls `loadMeasurer`
2. `loadMeasurer` calls `setState({measurer})` which causes a `render` + `componentDidUpdate`
3. `componentDidUpdate` calls `measure`
4. `measure` calls `setState({ strength })` which triggers a `render` + `componentDidUpdate`, which triggers another `measure` (is fine, no infinite loop prevented by `PureComponent`)
5. Second arg to `loadMeasurer` calls `measure` again.

It's fine, not a performance concern, just thinking perhaps this thing could be overall simpler? e.g.

```js
class PasswordStrength extends PureComponent<Props, State> {
    state = {
        strength: -1,
    }

    componentDidMount() {
        this.measure()
    }

    componentDidUpdate() {
        this.measure()
    }

    componentWillUnmount() {
        this.unmounted = true
    }

    unmounted: boolean

    measure = async () => {
        const { enabled, value } = this.props
        if (!enabled || !value) { return }

        const measurer = await zxcvbn()
        if (this.unmounted) { return }
        this.setState({
            strength: [0, 1, 1, 2, 2][measurer(value || '').score],
        })
    }

    render() {
        const { children, enabled, value } = this.props
        const strength = (enabled && value) ? this.state.strength : -1
        return children ? children(strength) : null
    }
}
```
Your call, just a suggestion.